src/service/action-impl.js

```diff
@@ -234,8 +263,9 @@ export class ActionService {
         const element = dev().assertElement(event.target);
         const keyCode = event.keyCode;
         if (keyCode == KeyCodes.ENTER || keyCode == KeyCodes.SPACE) {
-          if (!event.defaultPrevented &&
-              element.getAttribute('role') == 'button') {
+          const isTapWidgetRole = TAP_WIDGET_ROLES.includes(
```

Comment:
Hm. I'm a little concerned about this being O(n) since it executes on every action, but maybe I'm grasping at straws since `TAP_WIDGET_ROLES` is small.

Is it worth considering making TAP_WIDGET_ROLES a simple map and then check for key so it's O(1)?

Reply:
Good call, that sounds more future-proof.